luthier/src/main/java/com/yahoo/luthier/webservice/data/config/dimension/ExternalDimensionsLoader.java

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.luthier.webservice.data.config.dimension;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.joda.JodaModule;
+import com.yahoo.bard.webservice.data.config.dimension.DefaultKeyValueStoreDimensionConfig;
+import com.yahoo.bard.webservice.data.config.dimension.DimensionConfig;
+import com.yahoo.bard.webservice.data.dimension.KeyValueStore;
+import com.yahoo.bard.webservice.data.dimension.MapStoreManager;
+import com.yahoo.bard.webservice.data.dimension.SearchProvider;
+import com.yahoo.bard.webservice.data.dimension.impl.ScanSearchProviderManager;
+import com.yahoo.bard.webservice.util.StreamUtils;
+import com.yahoo.luthier.webservice.data.config.ExternalConfigLoader;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Hold all the dimension configurations for the sample Bard instance.
+ */
+public class ExternalDimensionsLoader {
+
+    private final Set<DimensionConfig> dimensionConfigs;
+
+    /**
+     * Construct the dimension configurations.
+     *
+     * @param dimensionConfigLoader The external configuration loader for loading dimensions
+     * @param dimensionConfigFilePath The external file's url containing the external config information
+     */
+    public ExternalDimensionsLoader(
+            ExternalConfigLoader dimensionConfigLoader,
+            String dimensionConfigFilePath
+    ) {
+
+        JodaModule jodaModule = bindTemplates();
+        ObjectMapper objectMapper = new ObjectMapper().registerModule(jodaModule);
+
+        ExternalDimensionConfigTemplate dimensionConfig =
+                dimensionConfigLoader.parseExternalFile(
+                        dimensionConfigFilePath + "DimensionConfig.json",
+                        ExternalDimensionConfigTemplate.class, objectMapper);
+
+        this.dimensionConfigs = Collections.unmodifiableSet(
+                dimensionConfig.getDimensions().stream()
+                        .map(
+                                dimensionName -> new DefaultKeyValueStoreDimensionConfig(
+                                        dimensionName.build(),
+                                        dimensionName.getApiName(),
+                                        dimensionName.getDescription(),
+                                        dimensionName.getLongName(),
+                                        dimensionName.getCategory(),
+                                        dimensionName.getFields(dimensionConfig.getFieldSets()),
+                                        getDefaultKeyValueStore(dimensionName),
+                                        getDefaultSearchProvider(dimensionName)
```

Comment:
once build is changed, I think you should just be able to call build here instead of creating the DimensionConfig